qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -173,32 +178,36 @@ def show_preview(self):
         self.groupBoxPreview.show()
 
     def download_resource(self):
+        # pydevd_pycharm.settrace('127.0.0.1', port=53100, stdoutToServer=True, stderrToServer=True)
         resource = self.selected_resource()
-        file_path = download_resource_file(
-            resource.file, resource.resource_type, resource.uuid
-        )
-        if file_path:
-            text = f"Successfully download {resource.name} to {file_path}"
-            self.iface.messageBar().pushMessage(self.tr("Success"), text, duration=5)
-            if self.checkBoxOpenDirectory.isChecked():
-                QDesktopServices.openUrl(QUrl.fromLocalFile(str(file_path.parent)))
-        else:
-            text = f"Failed download {resource.name} from {resource.file}"
-            self.iface.messageBar().pushMessage(
-                self.tr("Warning"), text, level=Qgis.Warning, duration=5
+
+        # Show file dialog to select the download directory
+        directory = QFileDialog.getExistingDirectory(self, "Select Directory")
+        if directory:
+            file_path = download_resource_file(
+                resource.file, resource.resource_type, resource.uuid, directory
             )
+            if file_path:
+                text = f"Successfully download {resource.name} to {file_path}"
```

Comment:
f-string is good, but as far as I know, it doesn't support translation well. Unless there is an update about it.

Reply:
Replaced the use of f-string